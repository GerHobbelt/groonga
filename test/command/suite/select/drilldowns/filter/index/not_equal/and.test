table_create Tags TABLE_PAT_KEY ShortText
column_create Tags is_stop_word COLUMN_SCALAR Bool

table_create Booleans TABLE_PAT_KEY Bool
column_create Booleans index COLUMN_INDEX Tags is_stop_word

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos tag COLUMN_SCALAR Tags

load --table Memos
[
{"_key": "Groonga is fast!", "tag": "Groonga"},
{"_key": "Mroonga is fast!", "tag": "Mroonga"},
{"_key": "Groonga sticker!", "tag": "Groonga"},
{"_key": "Rroonga is fast!", "tag": "Rroonga"}
]

load --table Tags
[
{"_key": "Mroonga", "is_stop_word": true}
]

log_level --level info
#@add-important-log-levels info
#@add-ignore-log-pattern /\A\[io\]/
select Memos \
  --drilldowns[tag].keys tag \
  --drilldowns[tag].filter '_key != "Groonga" && is_stop_word != true' \
  --drilldowns[tag].output_columns _key,_nsubrecs,is_stop_word
#@remove-ignore-log-pattern /\A\[io\]/
#@remove-important-log-levels info
log_level --level notice
