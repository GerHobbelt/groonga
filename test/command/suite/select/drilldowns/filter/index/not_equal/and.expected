table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
column_create Tags is_stop_word COLUMN_SCALAR Bool
[[0,0.0,0.0],true]
table_create Booleans TABLE_PAT_KEY Bool
[[0,0.0,0.0],true]
column_create Booleans index COLUMN_INDEX Tags is_stop_word
[[0,0.0,0.0],true]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
load --table Memos
[
{"_key": "Groonga is fast!", "tag": "Groonga"},
{"_key": "Mroonga is fast!", "tag": "Mroonga"},
{"_key": "Groonga sticker!", "tag": "Groonga"},
{"_key": "Rroonga is fast!", "tag": "Rroonga"}
]
[[0,0.0,0.0],4]
load --table Tags
[
{"_key": "Mroonga", "is_stop_word": true}
]
[[0,0.0,0.0],1]
log_level --level info
[[0,0.0,0.0],true]
select Memos   --drilldowns[tag].keys tag   --drilldowns[tag].filter '_key != "Groonga" && is_stop_word != true'   --drilldowns[tag].output_columns _key,_nsubrecs,is_stop_word
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        4
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "tag",
          "Tags"
        ]
      ],
      [
        1,
        "Groonga is fast!",
        "Groonga"
      ],
      [
        2,
        "Mroonga is fast!",
        "Mroonga"
      ],
      [
        3,
        "Groonga sticker!",
        "Groonga"
      ],
      [
        4,
        "Rroonga is fast!",
        "Rroonga"
      ]
    ],
    {
      "tag": [
        [
          1
        ],
        [
          [
            "_key",
            "ShortText"
          ],
          [
            "_nsubrecs",
            "Int32"
          ],
          [
            "is_stop_word",
            "Bool"
          ]
        ],
        [
          "Rroonga",
          1,
          false
        ]
      ]
    }
  ]
]
log_level --level notice
[[0,0.0,0.0],true]
